packages/launchpad-ledger/src/launchpadledger.ts

```diff
@@ -125,82 +93,20 @@ export class LaunchpadLedger {
   }
 
   public async disconnect(): Promise<void> {
-    if (this.connectedApp) {
-      await this.connectedApp.transport.close();
-      this.connectedApp = null;
-    }
-  }
-
-  private async ensureConnected(): Promise<void> {
-    // assume good connection if connected once
-    if (this.connectedApp) {
-      return;
-    }
-
-    if (this.platform !== "node") {
-      verifyBrowserIsSupported(this.platform, this.userAgent);
-    }
-
-    const transport = await this.createTransport(defaultOpenTimeout);
-    this.connectedApp = {
-      transport: transport,
-      app: new CosmosApp(transport),
-    };
-
-    await this.verifyDeviceIsReady();
-  }
-
-  /**
-   * @param openTimeout The time to establish a connection in milliseconds. This is
-   *                    [passed into as the second argument into Transport.open](https://github.com/LedgerHQ/ledgerjs/blob/v5.25.2/packages/hw-transport/src/Transport.js#L235),
-   *                    which is ignored by both [TransportWebUSB.open](https://github.com/LedgerHQ/ledgerjs/blob/v5.25.2/packages/hw-transport-webusb/src/TransportWebUSB.js#L116)
-   *                    and [TransportNodeHid.open](https://github.com/LedgerHQ/ledgerjs/blob/v5.25.2/packages/hw-transport-node-hid/src/TransportNodeHid.js#L115).
-   */
-  private async createTransport(openTimeout: number): Promise<Transport> {
-    // HACK: Use a variable to get webpack to ignore this
-    const nodeJsTransportPackageName = "@ledgerhq/hw-transport-node-hid";
-    /* eslint-disable-next-line @typescript-eslint/naming-convention */
-    const { default: TransportClass } =
-      this.platform === "node"
-        ? await import(nodeJsTransportPackageName)
-        : await import("@ledgerhq/hw-transport-webusb");
-
-    try {
-      const transport = await TransportClass.create(openTimeout);
-      return transport;
-    } catch (error) {
-      const trimmedErrorMessage = error.message.trim();
-      if (trimmedErrorMessage.startsWith("No WebUSB interface found for your Ledger device")) {
-        throw new Error(
-          "Could not connect to a Ledger device. Please use Ledger Live to upgrade the Ledger firmware to version 1.5.5 or later.",
-        );
-      }
-      if (trimmedErrorMessage.startsWith("Unable to claim interface")) {
-        throw new Error("Could not access Ledger device. Is it being used in another tab?");
-      }
-      if (trimmedErrorMessage.startsWith("Not supported")) {
-        throw new Error(
-          "Your browser does not seem to support WebUSB yet. Try updating it to the latest version.",
-        );
-      }
-      if (trimmedErrorMessage.startsWith("No device selected")) {
-        throw new Error(
-          "You did not select a Ledger device. If you did not see your Ledger, check if the Ledger is plugged in and unlocked.",
-        );
-      }
-
-      throw error;
+    if (this.app) {
+      await this.transport.close();
+      this.app = null;
```

Comment:
When the transport is provided externally, do we even want to keep this methis? We could close the transport in the caller code for symmetry.